### ISSUE TITLE
[TOPIC-GPIO] gpi_api_1pin test failures

### ISSUE THREAD

**Issue by pabigot** [zephyrproject-rtos/zephyr#19692]
>     * `gpio_api_1pin`: Failed all tested boards so far. Investigating this is my next task

There is one issue I've noticed just recently that may cause 1 pin testcase to fail. The testcases testing the new port functions attempt to verify that the driver doesn't change other pins than the one that was requested. But the test doesn't take into account that the LED0 pin it's using for testing may be on the same port as UART used for the console. When reading port values the pin serving as UART RX will introduce inconsistency to the port value the testcase is expecting. If you see failures in any of `test_gpio_port_*` tests please ignore it for now. I'll provide a fix.

_Originally posted by @mnkp in https://github.com/zephyrproject-rtos/zephyr/pull/19579#issuecomment-539418063_

**Comment by pabigot** [comment]
Added issue for visibility.

#19379 for SX1509B also fails all `test_gpio_port_set_*` cases.  This case is a little more subtle and interesting: the nrf52_pca20020 Thingy:52 board has a RGB lightwell LEDs that are very bright, and RGB sense LEDs.  When the test green LED is illuminated the sense green LED goes high, so two pins on the port change in parallel.

```
FAIL - test_gpio_port_set_masked_get_raw
FAIL - test_gpio_port_set_masked_get
FAIL - test_gpio_port_set_masked_get_active_high
FAIL - test_gpio_port_set_masked_get_active_low
FAIL - test_gpio_port_set_bits_clear_bits_raw
FAIL - test_gpio_port_set_bits_clear_bits
FAIL - test_gpio_port_set_clr_bits_raw
FAIL - test_gpio_port_set_clr_bits
FAIL - test_gpio_port_toggle
```

**Comment by pabigot** [comment]
Should be closed by #19782
